internal/controller/controllers/preprovisioningimage_controller.go

```diff
@@ -180,6 +192,17 @@ func (r *PreprovisioningImageReconciler) setImage(generation int64, status *meta
 	return nil
 }
 
+func setNotCreatedCondition(image *metal3_v1alpha1.PreprovisioningImage) {
+	message := "Waiting for InfraEnv image to be created"
+	reason := imageConditionReason(strcase.ToCamel(message))
+	setImageCondition(image.GetGeneration(), &image.Status,
+		metal3_v1alpha1.ConditionImageReady, metav1.ConditionFalse,
+		reason, message)
+	setImageCondition(image.GetGeneration(), &image.Status,
```

Comment:
why do we need both `metal3_v1alpha1.ConditionImageReady` and `metal3_v1alpha1.ConditionImageError`?

Reply:
You are asking why we need to update it here or why do we need the ConditionImageError in general?

Reply:
The image could not be ready, but also not be encountering an error as in this case and the cooldown case.
Or the image could not be ready and there could be an error that the user needs to handle.

We need to be able to express both situations clearly.